web/yo/app/scripts/controllers/gene.js

```diff
@@ -520,79 +546,96 @@ angular.module('oncokbApp')
                         }
                     }
                 }
-            }
-
+            };
+            $scope.acceptChangesByPerson = function(userName) {
+                if(!userName) {
+                    dialogs.error('Error', 'Can not accept changes from invalid user name. Please contact the developer.');
+                    return false;
+                }
+                if ($scope.status.isDesiredGene) {
+                    $scope.status[userName].savingAll = true;
+                }
+                collectChangesByPerson(userName);
+                var apiCalls = [];
+                if(!_.isEmpty(evidencesAllUsers[userName].geneTypeEvidence)) {
+                    apiCalls.push(geneTypeUpdate(userName));
+                }
+                if(!_.isEmpty(evidencesAllUsers[userName].updatedEvidences)) {
+                    apiCalls.push(evidenceBatchUpdate(userName));
+                }
+                if(!_.isEmpty(evidencesAllUsers[userName].deletedEvidences)) {
+                    apiCalls.push(evidenceDeleteUpdate(userName));
+                }
+                $q.all(apiCalls)
+                    .then(function(result) {
+                        $scope.status[userName].savingAll = false;
+                        $scope.status[userName].noChanges = true;
+                        evidencesAllUsers[userName] = {};
+                    }, function(error) {
+                        // Errors already been handled seperatly in the single api calls.
+                        console.log('Error happened ', error);
+                    });
+            };
             function collectChangesByPerson(userName) {
-                $scope.status.savingAll = false;
-                $scope.status.noChanges = false;
-                myUpdatedEvidences = {};
-                myDeletedEvidences = [];
-                myGeneTypeEvidence = {};
-                myUpdatedEvidenceModels = [];
+                if(!userName) {
+                    console.log('Can not collect changes from invalid user name. Please double check the userName.');
```

Comment:
In the current implementation, it is unnecessary for me to do userName check in collectChangesByPerson since it won't be called if userName is invalid. Maybe I should remove it.